scripts/startup.sh

```diff
@@ -21,227 +23,48 @@ dualityd init --chain-id $CHAIN_ID duality
-    then
-        echo "Found Duality chain: $found_network"
-    else
-        echo "Could not establish connection to Duality chain, found network: ${found_network-none}"
-        echo "Exiting..."
+        echo "Cannot load genesis.json from: $GENESIS_FILE_URL"
         exit 1
     fi
 
-    # set chain settings
-    sed -i 's#persistent_peers = ""#persistent_peers = "'"$persistent_peers"'"#' /root/.duality/config/config.toml
-
-    # check if this node intends to become a validator
-    if [[ "$STARTUP_MODE" == "validator" && ! -z "$MNEMONIC" ]]
-    then
-
-        echo "Starting future validator fullnode..."
-        dualityd --log_level ${LOG_LEVEL:-info} start & :
-
-        # wait for node to finish catching up to the chain's current height
-        sleep 5
-        chain_block_height=$(echo $node_status_json | jq -r ".result.sync_info.latest_block_height")
-        node_status_json=$( dualityd status )
-        while [[ $( echo $node_status_json | jq .SyncInfo.catching_up ) == true ]]
-        do
-            node_block_height=$( echo $node_status_json | jq -r .SyncInfo.latest_block_height )
-            echo "Node is catching up to chain height... ~$( printf '%.1f\n' $( echo "100*$node_block_height/$chain_block_height" | bc -l ) )% done"
-            sleep 10
-            node_status_json=$( dualityd status )
-        done
-        echo "Node has caught up to chain height"
+# Use network genesis file if present
+elif [ -f $genesis_file ] && [ $STARTUP_MODE == "new" ]; then
+    echo "Using network genesis.json: $genesis_file"
+    cp $genesis_file ${HOME}/.duality/
 
-        # add validator key (--no-backup ensures the terminal from seeing/logging the MNEMONIC value)
-        echo -n "$MNEMONIC" | dualityd keys add validator --recover --no-backup --keyring-backend test
+# Fallback to auto-generated genesis.json
+elif [ $STARTUP_MODE == "new" ]; then
+    echo "Using auto-generated genesis file"
+fi
 
-        # sent request to become a validator (to the first RPC address defined)
-        dualityd tx staking create-validator \
-            --node-id `dualityd tendermint show-node-id` \
-            --chain-id $CHAIN_ID \
-            --pubkey `dualityd tendermint show-validator` \
-            --commission-rate="${VALIDATOR_COMMISSION_RATE:-1.0}" \
-            --commission-max-rate="${VALIDATOR_COMMISSION_MAX_RATE:-1.0}" \
-            --commission-max-change-rate="${VALIDATOR_COMMISSION_MAX_CHANGE_RATE:-1.0}" \
-            --min-self-delegation="${VALIDATOR_MIN_SELF_DELEGATION:-1}" \
-            --gas="${VALIDATOR_GAS:-auto}" \
-            --amount "${VALIDATOR_AMOUNT:-1000000stake}" \
-            --fees "${VALIDATOR_FEES:-0token}" \
-            --keyring-backend test \
-            --from validator -y
+# Start or join a chain
+# note: "new" should not be used for mainnet chains
+# for mainnets a custom genesis file should be curated outside of these scripts
+if [ $STARTUP_MODE == "new" ]
+then
```

Comment:
tiny formatting difference, all the other `if` statements nearby use `; then` statement on the same line.